### PR TITLE
feat: updates event and cloudevent interfaces

### PR DESCRIPTION
**Pull request by grant**
- Updates the function interfaces for the Node FF
  - Adds a `LegacyEvent` interface
  - Updates `CloudEvent` interface to CE 1.0 (must have been <1.0 with `schemaurl`)

### CODE REVIEW

File: src/functions.ts [hunk 1]
```diff
@@ -62,7 +75,7 @@ export interface CloudFunctionsContext {
   /**
    * The resource that emitted the event.
    */
-  resource?: string;
+  resource?: string | object;
```
Comment (matthewrobertson) [inline src/functions.ts:78]: 👍 I noticed this was an issue in my last PR but was reluctant to touch it...

File: src/functions.ts [hunk 2]
```diff
@@ -40,6 +40,19 @@ export type HandlerFunction =
...
+ */
+export interface LegacyEvent {
+  data: object;
+  context: CloudFunctionsContext;
+}
+
+interface Data {
+  data: object;
+}
+export type LegacyCloudFunctionsContext = CloudFunctionsContext | Data;
```
Comment (matthewrobertson) [inline src/functions.ts:54]: Is the `|` here because there are scenarios where the context argument to a user's function a `Data`?

**Review by matthewrobertson — APPROVED** [review]
LGTM

File: src/functions.ts (hunk 1 above)
Comment (grant) [reply src/functions.ts:78]: 👌🏻

File: src/functions.ts (hunk 2 above)
Comment (grant) [reply src/functions.ts:54]: Correct, iirc there were certain cases where the context argument was a data object.

**Pull request merged**